src/appleseed/renderer/modeling/project/projectfileupdater.cpp

```diff
@@ -1585,6 +1585,47 @@ namespace
             }
         }
     };
+
+    //
+    // Update from revision 21 to revision 22.
+    //
+    
+    class UpdateFromRevision_21
+      : public Updater
+    {
+      public:
+        explicit UpdateFromRevision_21(Project& project)
+          : Updater(project, 21)
+        {
+        }
+
+        void update() override
+        {
+            if (Scene* scene = m_project.get_scene())
+            {
+                for (each<CameraContainer> i = cameras; i; ++i)
+                {
+                    Dictionary& camera_params = i->get_parameters();
+
+                    copy_in_same_path(
+                        camera_params,
+                        "shutter_open_end_time",
+                        "shutter_open_time");
+
+                    copy_in_same_path(
+                        camera_params,
+                        "shutter_close_start_time",
+                        "shutter_close_time");
+                }
+            }
+        }
+
+      private:        
+        void copy_in_same_path(Dictionary& dict, const char* dest_key, const char* src_key){
```

Comment:
I think it can be defined in `Updater` class..

Reply:
Yes it makes sense to move it next to the other similar methods.

The name is not very clear, maybe `copy_if_exist_no_overwrite()`?

Also make static, and move the opening brace to its own line.